content/blog/2019/08/2019-08-26-jenkins-cli.adoc

```diff
@@ -0,0 +1,96 @@
+---
+layout: post
+title: "Managing Jenkins with jcli"
+tags:
+- general
+- jcli
+author: linuxsuren
+---
+
+As a developer, I usually use Jenkins like this:
+
+* Find a job which is related with my current work
+* Trigger that job
+* Check the output of log
+
+Sometimes, I might need to check the update center. Maybe a new plugin is needed, or I need to update an existing plugin.
+Or, I want to upload a plugin from my computer. For all these cases, I just don't need a UI or even a browser.
+I like to use a CLI to complete most of my tasks. For example, I use `kubectl` to manage my Kubernetes cluster,
+to create or modify the kubernetes resources. So, I start to think, 'Why not use a CLI to manage my Jenkins?'.
+
+== Why create a new one?
+
+First, I almost forgot about the existing Jenkins CLI, written in Java. Let me introduce how to use that one.
+
+Visit Jenkins page from `http://localhost:8080/jenkins/cli/`. You'll see a command like `java -jar jenkins-cli.jar -s http://localhost:8080/jenkins/ help`. So, a jar file needs to be download.
+We can use this command to complete this task `wget http://localhost:8080/jenkins/jnlpJars/jenkins-cli.jar`.
+
+Now you can see that this is not a Linux-style CLI. Please consider some points below:
+
+* The users must have a JRE. This is not convenient for developers who don't use Java.
+* Type more words. We always need to type `java -jar jenkins-cli.jar -s http://localhost:8080/jenkins/` as the initial command.
```

Comment:
```suggestion
* The CLI is too wordy. We always need to type `java -jar jenkins-cli.jar -s http://localhost:8080/jenkins/` as the initial command.
```

or so. Otherwise it is not a complete sentence